### ISSUE TITLE
All components should support UIContentSizeCategoryAdjusting behavior

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#8595]
## Feature Request/Suggestion
All components should support `adjustsFontForContentSizeCategory` APIs appropriate for the implementation. Clients should be able to set a `-[UIFont preferredFont...]` API font, set `adjustsFontForContentSizeCategory = YES` on the appropriate view or view controller, and see the fonts respond automatically for Dynamic Type.
